epam_tr/src/main/java/com/epam/tr/service/SecurityUserDetailService.java

```diff
@@ -0,0 +1,39 @@
+package com.epam.tr.service;
+
+import com.epam.tr.dao.Dao;
+import com.epam.tr.entities.AppUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+
+@Component
+public class SecurityUserDetailService implements UserDetailsService {
+
+    private static final String USER_NOT_FOUND = "User not found";
+    private Dao<AppUser> dao;
+
+    @Autowired
+    public SecurityUserDetailService(Dao<AppUser> dao) {
```

Comment:
uniformity.
you usually autowire beans like a filed, no?
then why do you that via a constructor here?

Reply:
Just use one approach: autowire via constructor or field.